mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -1306,4 +1306,33 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
    * satisfies this requirement, we simply use a type alias here.
    */
   private[recommendation] type ALSPartitioner = org.apache.spark.HashPartitioner
+
+  /**
+   * Private function to checkpoint the RDD and clean up its all of its parents' shuffles eagerly.
+   */
+  private[spark] def checkpointAndCleanParents[T](rdd: RDD[T], blocking: Boolean = false): Unit = {
+    val sc = rdd.sparkContext
+    // If there is no reference tracking we skip clean up.
+    if (sc.cleaner.isEmpty) {
+      return rdd.checkpoint()
```

Comment:
Ah thats a good catch (this used to not be an issue since I left the materilization in the initial PR for both). Anyways I'll refactor this to break up the cleanup and explicitly capture the deps.